File: cakk-api/src/main/java/com/cakk/api/utils/CustomSpringExpressionLanguageParser.java
package com.cakk.api.utils;

import org.springframework.expression.ExpressionParser;
import org.springframework.expression.spel.standard.SpelExpressionParser;
import org.springframework.expression.spel.support.StandardEvaluationContext;

import lombok.AccessLevel;
import lombok.NoArgsConstructor;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class CustomSpringExpressionLanguageParser {

	public static Object getDynamicValue(String spEL, String[] parameterNames, Object[] args) {
		ExpressionParser parser = new SpelExpressionParser();
		StandardEvaluationContext context = new StandardEvaluationContext();

		for (int i = 0; i < parameterNames.length; i++) {
			context.setVariable(parameterNames[i], args[i]);
		}

		return parser.parseExpression(spEL).getValue(context, Object.class);
	}
}

File: cakk-core/src/main/kotlin/com/cakk/core/utils/CustomSpringExpressionLanguageParser.kt
package com.cakk.core.utils

import org.springframework.expression.ExpressionParser
import org.springframework.expression.spel.standard.SpelExpressionParser
import org.springframework.expression.spel.support.StandardEvaluationContext

object CustomSpringExpressionLanguageParser {

    fun getDynamicValue(spEL: String, parameterNames: Array<String>, args: Array<Any>): Any? {
        val parser: ExpressionParser = SpelExpressionParser()
        val context = StandardEvaluationContext()

        for (i in parameterNames.indices) {
            context.setVariable(parameterNames[i], args[i])
        }

        return parser.parseExpression(spEL).getValue(context, Any::class.java)
    }
}
